packages/engine/lib/stateful/src/lib.rs

```diff
@@ -0,0 +1,7 @@
+// TODO: DOC: Describing this crate containing stateful things like agents, context, etc. and the
```

Comment:
I think we agreed to do these as part of these PRs. Do you feel strongly about waiting on this one until we have more modules?

Reply:
As we don't know currently, what exactly will go into this crate, I'd leave the full documentation to the next PR. However, I added a documentation for the `fields` module.